### PR TITLE
Add AmazonTest class with corresponding Page Object Model implementation

### PR DESCRIPTION
**Pull request by Shake-Ghairbyan**
#### Add `AmazonTest` class to `src/test/java/`.
-  Apply `SoftAssert`, `WebDriver`, `ChromeDriver`, `@BeforeSuite`, `@BeforeClass`, ` @Test(priority)`, `@AfterClass`.

#### Add `amazon` package with `HomePage`, `AuthorSearchPage`, `AuthorPage` classes to `src/main/java/`.
 - Apply  `By`, `WebDriver`, `WebElement`, `ExpectedConditions` and `WebDriverWait`.

_P.S. I know that tests are not independent. I just wanted to separate them to have a clear logical flow. 
       In order not to have an issue with required `url` addresses at specified routes, applied ` @Test(priority)`._

### CODE REVIEW

File: src/main/java/amazon/AuthorPage.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        List<WebElement> pricesOfBooks = driver.findElements(priceOfBookLoc);
+        for (int i = 1; i < pricesOfBooks.size(); i++) {
+            double current = parseStringToDouble(pricesOfBooks.get(i).getText());
+            double previous = parseStringToDouble(pricesOfBooks.get(i - 1).getText());
+            if (previous > current) {
+                return false;
+            }
+        }
+        return true;
+    }
```
Comment (David91) [inline src/main/java/amazon/AuthorPage.java]: Whole logic and checks are in page object. It'd be better to return all the prices and then all the checks and assertions do in test method, so test method will be more informative and it'll report exactly at what point assertion happens if any.
In this case your test only will report that expected true, but actual is false ( if assertion fails ).

File: src/main/java/amazon/AuthorSearchPage.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    public WebElement authorNameIsClickable(String authorName) throws Exception {
+        List<WebElement> searchResultsAuthorLinks = driver.findElements(searchResultsAuthorLinksLoc);
+        for (WebElement e : searchResultsAuthorLinks) {
+            if (e.getText().toLowerCase().equals(authorName)) {
+                return e;
+            }
+        }
+        throw new Exception("Author name in search results is not clickable.");
```
Comment (David91) [inline src/main/java/amazon/AuthorSearchPage.java]: Your method's name doesn't match with what it does. It assumes to return boolean value, but returns WebElement.
And logic inside have nothing to do with method's name.

File: src/main/java/amazon/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        WebElement books = driver.findElement(navBarBooksLoc);
+        noSuchElement(books);
+        books.click();
+    }
+
+    private void noSuchElement(WebElement element) {
+        if (element == null) {
+            throw new RuntimeException("ERROR: No such element");
+        }
+    }
```
Comment (David91) [inline src/main/java/amazon/HomePage.java]: Why is this method ? If element won't be found, a runtime exception will be thrown anyway, so this method can't be executed.

File: src/test/java/AmazonTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        String actualBooksByAuthorText = authorPage.booksByAuthorText();
+        String assertMessageAuthor = "Author name in books by section field differs from expected one.";
+        softAssert.assertEquals(actualBooksByAuthorText, expectedBooksByAuthorText, assertMessageAuthor);
+
+        authorPage.sortByPriceLowToHigh();
+        authorPage.waitUntilPageLoads();
+
+        boolean assertCond = authorPage.areSortedByPriceLowToHigh();
+        softAssert.assertTrue(assertCond, "Books are not sorted from low to high by price.");
+        softAssert.assertAll();
```
Comment (David91) [inline src/test/java/AmazonTest.java]: The task that I've given is one flow, which has intermediate assertions. So keep all steps and assertions in 1 test method.

**Review by David91 — APPROVED** [review]

File: src/main/java/amazon/HomePage.java (hunk 3 above)
Comment (Shake-Ghairbyan) [reply src/main/java/amazon/HomePage.java]: Addressed in 4281fcc.

File: src/test/java/AmazonTest.java (hunk 4 above)
Comment (Shake-Ghairbyan) [reply src/test/java/AmazonTest.java]: Addressed in 80a2655.

File: src/main/java/amazon/AuthorPage.java (hunk 1 above)
Comment (Shake-Ghairbyan) [reply src/main/java/amazon/AuthorPage.java]: Can you elaborate? 
If I understand you right, for every pair I should write assertion in a loop in `AmazonTest`.  And in `AuthorPage` from pricesOfBooks generate another list of doubles with exact prices.

Comment (David91) [reply src/main/java/amazon/AuthorPage.java]: Correct.

Comment (Shake-Ghairbyan) [reply src/main/java/amazon/AuthorPage.java]: Addressed in 6fbd353.

File: src/main/java/amazon/AuthorSearchPage.java (hunk 2 above)
Comment (Shake-Ghairbyan) [reply src/main/java/amazon/AuthorSearchPage.java]: Addressed in 6fbd353.

**Review by David91 — APPROVED** [review]

**Pull request merged**